fix(ai): update options-param templating documentation (#8720)
Kong/docs.konghq.com: app/_hub/kong-inc/ai-proxy/how-to/_sdk-usage.md

* `$(uri_captures.name)`
* `$(query_params.name)`

> `name` is either the header name, URI named regular-expression capture (in the route path), or the query parameter name, respectively.

Reviewer: ```suggestion
`name` is either the header name, URI named regular expression capture (in the route path), or the query parameter name, respectively.
```

## Use case examples
### Use the Azure deployment relevant to a specific model name

You could use a header capture to insert the requested model name directly into the

> plugin configuration for an Azure-OpenAI Kong AI gateway deployment, as a string

Reviewer: ```suggestion
plugin configuration for Kong AI Gateway deployment with Azure OpenAI, as a string
```

substitution, using the following example:

```yaml
routes:
  - name: azure-chat-model-from-path
    paths:
      - "~/azure/.*"
    plugins:
      - name: ai-proxy
        config:
          route_type: "llm/v1/chat"
          auth:
            azure_use_managed_identity: true
          model:
            provider: "azure"
            model: "$(headers.x-model-name)"
            options:
              azure_instance: "llm-deployment-v1"
              azure_deployment_id: "west-europe-$(headers.x-model-name)"
              azure_api_version: "2024-10-01"
```

> You can now target Azure Model deployment names `west-europe-gpt-4o` with the following sample request:

Reviewer: ```suggestion
You can now target an Azure model deployment named `west-europe-gpt-4o` with the following sample request:
```

```sh
cat <<EOF > request.json
{
  "messages": [
    {
      "role": "user",
      "content": [
        {
          "type": "text",
          "text": "This is my question."
        }
      ]
    }
  ]
}
EOF

> curl -d @request.json -H "x-test: azure-chat-open-model-managed-identity" -H "x-model-name: gpt-4o" http://127.0.0.1:8000/1/chat/completions

Reviewer: ```suggestion
curl http://localhost:8000/1/chat/completions \
  -H "x-test: azure-chat-open-model-managed-identity" \
  -H "x-model-name: gpt-4o" \
  -d @request.json
```
standardizing the formatting
